tfe/data_source_outputs_test.go

```diff
@@ -0,0 +1,224 @@
+
+	client, err := getClientUsingEnv()
+	if err != nil {
+		t.Fatalf("error getting client %v", err)
+	}
+
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	fileName := "test-fixtures/state-versions/terraform-empty-outputs.tfstate"
+	orgName, wsName, orgCleanup, wsCleanup := createOutputs(t, client, rInt, fileName)
+	defer orgCleanup()
+	defer wsCleanup()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEOutputs_dataSource_emptyOutputs(rInt, orgName, wsName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"tfe_organization.foobar", "name", fmt.Sprintf("tst-%d", rInt)),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "name", fmt.Sprintf("workspace-test-%d", rInt)),
+					resource.TestCheckResourceAttr(
+						"data.tfe_outputs.foobar", "organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_outputs.foobar", "workspace", wsName),
+					// This is relies on test-fixtures/state-versions/terraform-empty-outputs.tfstate
+					testCheckOutputState("state_output", &terraform.OutputState{
+						Value: map[string]interface{}{},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOutputState(name string, expectedOutputState *terraform.OutputState) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ms := s.RootModule()
+		rs, ok := ms.Outputs[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+		if rs.String() != expectedOutputState.String() {
+			return fmt.Errorf("Expected the output state %s to match expected output state %s", rs.String(), expectedOutputState.String())
+		}
+		return nil
+	}
+}
+
+func createOutputs(t *testing.T, client *tfe.Client, rInt int, fileName string) (string, string, func(), func()) {
+	var orgCleanup func()
+	var wsCleanup func()
+
+	org, err := client.Organizations.Create(ctx, tfe.OrganizationCreateOptions{
+		Name:  tfe.String(fmt.Sprintf("tst-terraform-%d", rInt)),
+		Email: tfe.String(fmt.Sprintf("%d@tfe.local", rInt)),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	orgCleanup = func() {
+		if err := client.Organizations.Delete(ctx, org.Name); err != nil {
+			t.Errorf("Error destroying organization! WARNING: Dangling resources\n"+
+				"may exist! The full error is shown below.\n\n"+
+				"Organization: %s\nError: %s", org.Name, err)
+		}
+	}
+
+	ws, err := client.Workspaces.Create(ctx, org.Name, tfe.WorkspaceCreateOptions{
+		Name: tfe.String(fmt.Sprintf("tst-workspace-test-%d", rInt)),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	wsCleanup = func() {
+		if err := client.Workspaces.Delete(ctx, org.Name, ws.Name); err != nil {
+			t.Errorf("Error destroying workspace! WARNING: Dangling resources\n"+
+				"may exist! The full error is shown below.\n\n"+
+				"Workspace: %s\nError: %s", ws.Name, err)
+		}
+	}
```

Comment:
Do you need this? The organization cleanup should cascade and delete the workspace anyway.

Reply:
That's true. Will remove